src/test/scala/ing/wbaa/druid/dql/DQLSpec.scala

```diff
@@ -413,4 +429,118 @@ class DQLSpec extends WordSpec with Matchers with ScalaFutures {
     }
   }
 
+  "DQL also work with 'select' queries" should {
+    val resultsThreshold = 10
+    val numberOfResults  = resultsThreshold * 24
+
+    val query: SelectQuery = DQL
+      .select(resultsThreshold)
+      .dimensions(d"channel", d"cityName", d"countryIsoCode", d"user")
+      .granularity(GranularityType.Hour)
+      .interval("2011-06-01/2017-06-01")
+      .build()
+
+    "successfully be interpreted by Druid" in {
+
+      val request = query.execute()
+      whenReady(request) { response =>
+        val resultList = response.list[SelectResult]
+        resultList.size shouldBe numberOfResults
+
+        val resultSeries = response.series[SelectResult]
+        resultSeries.size shouldBe 24
+        resultSeries.map { case (_, results) => results.size }.sum shouldBe numberOfResults
+      }
+    }
+
+    "successfully be streamed" in {
+      val requestSeq = query.streamAs[SelectResult].runWith(Sink.seq)
+      whenReady(requestSeq) { response =>
+        response.size shouldBe numberOfResults
+      }
+
+      val requestSeries = query.streamSeriesAs[SelectResult].runWith(Sink.seq)
+      whenReady(requestSeries) { response =>
+        response.size shouldBe numberOfResults
+      }
```

Comment:
I don't know the details of how the Akka streaming stuff works, so open question: is the test response large enough compared to the buffer sizes that all the streaming stuff is truly exercised (i.e. it's more than a single-chunk response)?

Reply:
Good point.

Regarding the test, the granularity type of the testing query is hourly and the threshold of events for each result is 10. In practise the datasource contains data only for a single day `2015-09-12`. Therefore, the select query will result to an array of 24 JSON result structures (one for each hour) and each one will contain 10 events (due to threshold of 10). 

Streaming JSON is being performed on the top-level JSON result structures (24 in our test), therefore in situations that the threshold is large we may not going to have any benefit from streaming. Only when the top-level JSON structures are many (e.g., months of data and hourly granularity) we benefit from streaming.

In order to have streaming support also for nested structures, we need something like [akka-stream-alpakka-json-streaming](https://doc.akka.io/docs/alpakka/current/data-transformations/json.html#streaming-of-nested-structures). In fact this is something that I've tried to implement at the beginning, but I couldn't make it to work with the nested structure of Select response.